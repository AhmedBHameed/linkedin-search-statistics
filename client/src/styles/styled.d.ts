import 'styled-components';

/**
 * For more information whey we need this file @see https://styled-components.com/docs/api#create-a-declarations-file
 */
declare module 'styled-components' {
  export interface DefaultTheme {
    colors: {
      blue5: string;
      blue50: string;
      blue90: string;
      red: string;
      highlightText: string;
    };
  }
}
