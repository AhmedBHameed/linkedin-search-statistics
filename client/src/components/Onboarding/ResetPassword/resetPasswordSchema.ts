import {
  confirmPasswordWith,
  Joi,
  requiredPassword,
} from '../../../util/validations';
import {ResetPasswordModel} from './models/ResetPasswordModel';

const resetPasswordSchema = Joi.object<ResetPasswordModel>({
  newPassword: requiredPassword(),
  confirmPassword: confirmPasswordWith('newPassword'),
});

export default resetPasswordSchema;
