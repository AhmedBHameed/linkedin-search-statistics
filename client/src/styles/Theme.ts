/**
 * The number 50 is the scale of the color according to Figma
 * @see https://www.figma.com/file/WCipxn54SiYJSVpZ3uTUW5/BiomeDx-Styleguide-und-Report-%234?node-id=13%3A140
 */

const highlightText = '#ffc069';

const lightTheme = {
  red: 'red',
  blue5: '#e7f2fe',
  blue50: '#047CB3',
  blue90: '#011e2f',
  highlightText,
};

const darkTheme = {
  red: '#ff7777',
  blue5: '#e7f2fe',
  blue50: '#047CB3',
  blue90: '#011e2f',
  highlightText,
};

export type ThemeProps = typeof darkTheme;
export {lightTheme, darkTheme};
