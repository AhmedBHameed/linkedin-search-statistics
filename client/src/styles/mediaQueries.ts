export const sm = 576;
export const md = 768;
export const lg = 992;
export const xl = 1200;
export const xxl = 1600;

const devices = {
  smMediaQuery: `(min-width: ${sm}px)`,
  mdMediaQuery: `(min-width: ${md}px)`,
  lgMediaQuery: `(min-width: ${lg}px)`,
  xlMediaQuery: `(min-width: ${xl}px)`,
  xxlMediaQuery: `(min-width: ${xxl}px)`,
};

export default devices;
