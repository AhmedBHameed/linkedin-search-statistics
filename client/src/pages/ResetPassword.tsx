import {KeyOutlined, LoginOutlined} from '@ant-design/icons';
import {joiResolver} from '@hookform/resolvers/joi';
import {Card, Col, Row} from 'antd';
import React, {useCallback, useState} from 'react';
import {Controller, useForm} from 'react-hook-form';

import {Link, useParams} from 'react-router-dom';
import {FormControl, PasswordField} from '../components/Forms/Form';
import useNavigateToLogin from '../components/Onboarding/Login/hooks/navigateToLoginHook';
import PasswordHints from '../components/Onboarding/PasswordHints';
import {ResetPasswordModel} from '../components/Onboarding/ResetPassword/models/ResetPasswordModel';
import resetPasswordSchema from '../components/Onboarding/ResetPassword/resetPasswordSchema';
import auth from '../components/Onboarding/util/auth';
import {
  FormTitle,
  FullWidthLoadingButton,
  OnboardingForm,
  StyledBrand,
  StyledRow,
} from '../components/shared/onboardingStyles';
import useToastMessage from '../components/ToastMessage/hooks/toastMessageHook';
import NAVIGATION_ROUTES from '../config/NavigationRoutes';
import {callTryCatch} from '../util/callTryCatch';

type VerificationUserInput = {
  userId: string;
  verificationId: string;
};

const ResetPassword: React.FC = () => {
  const [loading, setLoading] = useState(false);
  const {notifyError, notifySuccess} = useToastMessage();
  const params = useParams<VerificationUserInput>();
  const {goToLogin} = useNavigateToLogin();

  const {
    control,
    handleSubmit,
    formState: {errors},
  } = useForm<ResetPasswordModel>({
    resolver: joiResolver(resetPasswordSchema),
    defaultValues: {
      newPassword: '',
      confirmPassword: '',
    },
  });

  const resetPassword = useCallback(
    async (formData: ResetPasswordModel) => {
      setLoading(true);
      const [error] = await callTryCatch(async () =>
        auth.resetPassword({
          newPassword: formData.newPassword,
          userId: params.userId,
          verificationId: params.verificationId,
        })
      );

      setLoading(false);

      if (error) {
        // eslint-disable-next-line no-console
        console.error('<ResetPassword />', error);
        notifyError({
          message: 'Reset password failed',
          description: 'Oops! Something went wrong while resetting password',
        });

        return;
      }

      notifySuccess({
        message: 'Password reset',
        description: 'Great! you password has been reset',
      });
      goToLogin();
    },
    [params, notifyError, notifySuccess, goToLogin]
  );

  const newPasswordError = errors.newPassword?.message;
  const confirmPasswordError = errors.confirmPassword?.message;

  return (
    <StyledRow align="middle" justify="center">
      <Col lg={8} md={10} sm={12} xs={24}>
        <OnboardingForm
          onSubmit={handleSubmit(resetPassword)}
          testId="reset-password-form"
        >
          <Card>
            <Row align="middle" justify="center">
              <StyledBrand />
            </Row>

            <Row>
              <Col xs={24}>
                <FormTitle level={5}>Reset password</FormTitle>
              </Col>
            </Row>

            <Row>
              <Col xs={24}>
                <FormControl error={newPasswordError} label="Reset password">
                  <Controller
                    control={control}
                    name="newPassword"
                    render={({field: {value, onChange}}) => (
                      <PasswordField
                        ariaLabel="new password"
                        error={!!newPasswordError}
                        name="newPassword"
                        onChange={onChange}
                        placeholder="New password"
                        prefixIcon={<KeyOutlined />}
                        size="large"
                        testId="new-password-input"
                        value={value}
                      />
                    )}
                  />
                </FormControl>
              </Col>
            </Row>

            <Row>
              <Col xs={24}>
                <FormControl
                  error={confirmPasswordError}
                  label="Confirm password"
                >
                  <Controller
                    control={control}
                    name="confirmPassword"
                    render={({field: {value, onChange}}) => (
                      <PasswordField
                        ariaLabel="confirm password"
                        error={!!confirmPasswordError}
                        name="confirmPassword"
                        onChange={onChange}
                        placeholder="Confirm password"
                        prefixIcon={<KeyOutlined />}
                        size="large"
                        testId="confirm-password-input"
                        value={value}
                      />
                    )}
                  />
                </FormControl>
              </Col>
            </Row>

            <Col xs={24}>
              {errors.newPassword?.message && <PasswordHints />}
            </Col>

            <Row justify="end">
              <Link to={NAVIGATION_ROUTES.login.path}>Return to login</Link>
            </Row>

            <Row>
              <FullWidthLoadingButton
                disabled={loading}
                htmlType="submit"
                icon={<LoginOutlined />}
                loading={loading}
                size="large"
              >
                Reset
              </FullWidthLoadingButton>
            </Row>

            <Row align="middle" justify="center">
              <Link to={NAVIGATION_ROUTES.signup.path}>
                Don&apos;t have account?
              </Link>
            </Row>
          </Card>
        </OnboardingForm>
      </Col>
    </StyledRow>
  );
};

export default ResetPassword;
