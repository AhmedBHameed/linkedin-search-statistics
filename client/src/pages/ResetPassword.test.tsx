import {act, fireEvent, render, waitFor} from '@testing-library/react';
import {Space} from 'antd';

import {BrowserRouter as Router} from 'react-router-dom';
import {ThemeProvider} from 'styled-components';

import {mockConsoleErrorFun, mockMediaQuery} from '../setupTests';
import {md} from '../styles/mediaQueries';
import {lightTheme} from '../styles/Theme';
import ResetPassword from './ResetPassword';

mockConsoleErrorFun();

const mockNotifyError = jest.fn();
const mockNotifySuccess = jest.fn();
jest.mock('../components/ToastMessage/hooks/toastMessageHook', () => () => ({
  notifyError: mockNotifyError,
  notifySuccess: mockNotifySuccess,
}));

const mockGoToLogin = jest.fn();
jest.mock('../components/Login/hooks/navigateToLoginHook', () => () => ({
  goToLogin: mockGoToLogin,
}));

const mockResetPasswordHook = jest.fn();
jest.mock('../components/ResetPassword/hooks/resetPasswordHook', () => () => ({
  handleResetPassword: mockResetPasswordHook,
}));

const MOCK_RESET_PASSWORD = {
  newPassword: 'Test@123123',
  confirmPassword: 'Test@123123',
};

function renderComponent() {
  mockMediaQuery(md, md);

  return render(
    <ThemeProvider theme={{colors: lightTheme}}>
      <Router>
        <Space>
          <ResetPassword />
        </Space>
      </Router>
    </ThemeProvider>
  );
}

describe('ResetPassword', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.clearAllMocks();
    jest.useRealTimers();
  });

  it('matches snapshot', () => {
    const {container} = renderComponent();

    expect(container).toMatchSnapshot();
  });

  describe('Form validation', () => {
    it('shows required helper text as a hint when submitting empty value.', async () => {
      const {findAllByTestId, getByTestId} = renderComponent();

      act(() => {
        fireEvent.submit(getByTestId('reset-password-form'));
      });

      const inputErrors = await findAllByTestId('input-error');

      expect(inputErrors[0].textContent).toBe('Field required');
      expect(mockResetPasswordHook).toHaveBeenCalledTimes(0);
      expect(mockGoToLogin).toHaveBeenCalledTimes(0);
      expect(mockNotifyError).toHaveBeenCalledTimes(0);
      expect(mockNotifySuccess).toHaveBeenCalledTimes(0);
    });

    it('shows password pattern helper text as a hint when submitting invalid password.', async () => {
      const {findAllByTestId, getByTestId} = renderComponent();

      await act(async () => {
        fireEvent.change(getByTestId('new-password-input'), {
          target: {value: 'invalidPassword'},
        });
        await waitFor(() => {});

        fireEvent.submit(getByTestId('reset-password-form'));
      });

      const inputErrors = await findAllByTestId('input-error');

      expect(inputErrors[0].textContent).toBe(
        'validationError.invalidPassword'
      );

      expect(getByTestId('password-hints')).toBeInTheDocument();
      expect(mockResetPasswordHook).toHaveBeenCalledTimes(0);
      expect(mockGoToLogin).toHaveBeenCalledTimes(0);
      expect(mockNotifyError).toHaveBeenCalledTimes(0);
      expect(mockNotifySuccess).toHaveBeenCalledTimes(0);
    });

    it("shows password does not match helper text as a hint when they don't match.", async () => {
      const {findAllByTestId, getByTestId} = renderComponent();

      await act(async () => {
        fireEvent.change(getByTestId('new-password-input'), {
          target: {value: MOCK_RESET_PASSWORD.newPassword},
        });
        await waitFor(() => {});

        fireEvent.change(getByTestId('confirm-password-input'), {
          target: {value: 'notMatchedPassword'},
        });
        await waitFor(() => {});

        fireEvent.submit(getByTestId('reset-password-form'));
      });

      const inputErrors = await findAllByTestId('input-error');

      expect(inputErrors[0].textContent).toBe(
        'validationError.passwordsNotMatch'
      );
      expect(mockResetPasswordHook).toHaveBeenCalledTimes(0);
      expect(mockGoToLogin).toHaveBeenCalledTimes(0);
      expect(mockNotifyError).toHaveBeenCalledTimes(0);
      expect(mockNotifySuccess).toHaveBeenCalledTimes(0);
    });

    it('submits valid data and show successful toast message.', async () => {
      const {getByTestId} = renderComponent();

      await act(async () => {
        fireEvent.change(getByTestId('new-password-input'), {
          target: {value: MOCK_RESET_PASSWORD.newPassword},
        });
        await waitFor(() => {});

        fireEvent.change(getByTestId('confirm-password-input'), {
          target: {value: MOCK_RESET_PASSWORD.confirmPassword},
        });
        await waitFor(() => {});

        fireEvent.submit(getByTestId('reset-password-form'));
      });

      expect(mockResetPasswordHook).toHaveBeenCalledTimes(1);
      expect(mockResetPasswordHook).toHaveBeenCalledWith(MOCK_RESET_PASSWORD);

      expect(mockGoToLogin).toHaveBeenCalledTimes(1);
      expect(mockGoToLogin).toHaveBeenCalledWith(/* Nothing */);

      expect(mockNotifyError).toHaveBeenCalledTimes(0);

      expect(mockNotifySuccess).toHaveBeenCalledTimes(1);
      expect(mockNotifySuccess).toHaveBeenCalledWith({
        message: 'resetPassword.success.passwordResetSubmitted.title',
        description: 'resetPassword.success.passwordResetSubmitted.message',
      });
    });

    it('triggers toast message when async throw an error', async () => {
      mockResetPasswordHook.mockRejectedValueOnce(new Error('async error'));

      const {getByTestId} = renderComponent();

      await act(async () => {
        fireEvent.change(getByTestId('new-password-input'), {
          target: {value: MOCK_RESET_PASSWORD.newPassword},
        });
        await waitFor(() => {});

        fireEvent.change(getByTestId('confirm-password-input'), {
          target: {value: MOCK_RESET_PASSWORD.confirmPassword},
        });
        await waitFor(() => {});

        fireEvent.submit(getByTestId('reset-password-form'));
      });

      expect(mockResetPasswordHook).toHaveBeenCalledTimes(1);
      expect(mockResetPasswordHook).toHaveBeenCalledWith(MOCK_RESET_PASSWORD);

      expect(mockGoToLogin).toHaveBeenCalledTimes(0);

      expect(mockNotifySuccess).toHaveBeenCalledTimes(0);

      expect(mockNotifyError).toHaveBeenCalledTimes(1);
      expect(mockNotifyError).toHaveBeenCalledWith({
        message: 'common.error.somethingWentWrong.title',
        description: 'common.error.somethingWentWrong.message',
      });
    });
  });
});
