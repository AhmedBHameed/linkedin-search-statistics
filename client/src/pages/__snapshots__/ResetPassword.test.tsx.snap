// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword matches snapshot 1`] = `
<div>
  <div
    class="ant-space ant-space-horizontal ant-space-align-center"
  >
    <div
      class="ant-space-item"
    >
      <div
        class="ant-row ant-row-center ant-row-middle sc-kEqXSa fUiPLR"
      >
        <div
          class="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-md-10 ant-col-lg-8"
        >
          <form
            class="sc-iqAclL kgxeLt"
            data-testid="reset-password-form"
          >
            <div
              class="ant-card ant-card-bordered"
            >
              <div
                class="ant-card-body"
              >
                <div
                  class="ant-row ant-row-center ant-row-middle"
                >
                  <svg
                    class="sc-bdnxRM goXlsq"
                  >
                    brand-light-theme.svg
                  </svg>
                </div>
                <div
                  class="ant-row"
                >
                  <div
                    class="ant-col ant-col-xs-24"
                  >
                    <h5
                      class="ant-typography sc-crzoAE bvPise"
                    >
                      resetPassword.headTitle
                    </h5>
                  </div>
                </div>
                <div
                  class="ant-row"
                >
                  <div
                    class="ant-col ant-col-xs-24"
                  >
                    <div
                      class="ant-row ant-form-item sc-eCApnc fExcEz"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class=""
                          title="resetPassword.newPasswordLabel"
                        >
                          resetPassword.newPasswordLabel
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <span
                              class="ant-input-affix-wrapper ant-input-affix-wrapper-lg ant-input-password ant-input-password-large"
                            >
                              <span
                                class="ant-input-prefix"
                              >
                                <span
                                  aria-label="key"
                                  class="anticon anticon-key"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="key"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M608 112c-167.9 0-304 136.1-304 304 0 70.3 23.9 135 63.9 186.5l-41.1 41.1-62.3-62.3a8.15 8.15 0 00-11.4 0l-39.8 39.8a8.15 8.15 0 000 11.4l62.3 62.3-44.9 44.9-62.3-62.3a8.15 8.15 0 00-11.4 0l-39.8 39.8a8.15 8.15 0 000 11.4l62.3 62.3-65.3 65.3a8.03 8.03 0 000 11.3l42.3 42.3c3.1 3.1 8.2 3.1 11.3 0l253.6-253.6A304.06 304.06 0 00608 720c167.9 0 304-136.1 304-304S775.9 112 608 112zm161.2 465.2C726.2 620.3 668.9 644 608 644c-60.9 0-118.2-23.7-161.2-66.8-43.1-43-66.8-100.3-66.8-161.2 0-60.9 23.7-118.2 66.8-161.2 43-43.1 100.3-66.8 161.2-66.8 60.9 0 118.2 23.7 161.2 66.8 43.1 43 66.8 100.3 66.8 161.2 0 60.9-23.7 118.2-66.8 161.2z"
                                    />
                                  </svg>
                                </span>
                              </span>
                              <input
                                action="click"
                                aria-label="new password"
                                class="ant-input ant-input-lg"
                                data-testid="new-password-input"
                                name="newPassword"
                                placeholder="resetPassword.newPasswordPlaceholder"
                                type="password"
                                value=""
                              />
                              <span
                                class="ant-input-suffix"
                              >
                                <span
                                  aria-label="eye-invisible"
                                  class="anticon anticon-eye-invisible ant-input-password-icon"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="eye-invisible"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                    />
                                    <path
                                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                            <small
                              class="sc-jSFjdj fmCvnz ant-form-item-explain ant-form-item-explain-error"
                              data-testid=""
                            >
                              Â 
                            </small>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row"
                >
                  <div
                    class="ant-col ant-col-xs-24"
                  >
                    <div
                      class="ant-row ant-form-item sc-eCApnc fExcEz"
                    >
                      <div
                        class="ant-col ant-form-item-label"
                      >
                        <label
                          class=""
                          title="resetPassword.confirmPasswordLabel"
                        >
                          resetPassword.confirmPasswordLabel
                        </label>
                      </div>
                      <div
                        class="ant-col ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <span
                              class="ant-input-affix-wrapper ant-input-affix-wrapper-lg ant-input-password ant-input-password-large"
                            >
                              <span
                                class="ant-input-prefix"
                              >
                                <span
                                  aria-label="key"
                                  class="anticon anticon-key"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="key"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M608 112c-167.9 0-304 136.1-304 304 0 70.3 23.9 135 63.9 186.5l-41.1 41.1-62.3-62.3a8.15 8.15 0 00-11.4 0l-39.8 39.8a8.15 8.15 0 000 11.4l62.3 62.3-44.9 44.9-62.3-62.3a8.15 8.15 0 00-11.4 0l-39.8 39.8a8.15 8.15 0 000 11.4l62.3 62.3-65.3 65.3a8.03 8.03 0 000 11.3l42.3 42.3c3.1 3.1 8.2 3.1 11.3 0l253.6-253.6A304.06 304.06 0 00608 720c167.9 0 304-136.1 304-304S775.9 112 608 112zm161.2 465.2C726.2 620.3 668.9 644 608 644c-60.9 0-118.2-23.7-161.2-66.8-43.1-43-66.8-100.3-66.8-161.2 0-60.9 23.7-118.2 66.8-161.2 43-43.1 100.3-66.8 161.2-66.8 60.9 0 118.2 23.7 161.2 66.8 43.1 43 66.8 100.3 66.8 161.2 0 60.9-23.7 118.2-66.8 161.2z"
                                    />
                                  </svg>
                                </span>
                              </span>
                              <input
                                action="click"
                                aria-label="confirm password"
                                class="ant-input ant-input-lg"
                                data-testid="confirm-password-input"
                                name="confirmPassword"
                                placeholder="resetPassword.confirmPasswordPlaceholder"
                                type="password"
                                value=""
                              />
                              <span
                                class="ant-input-suffix"
                              >
                                <span
                                  aria-label="eye-invisible"
                                  class="anticon anticon-eye-invisible ant-input-password-icon"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="eye-invisible"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                    />
                                    <path
                                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                            <small
                              class="sc-jSFjdj fmCvnz ant-form-item-explain ant-form-item-explain-error"
                              data-testid=""
                            >
                              Â 
                            </small>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-col ant-col-xs-24"
                />
                <div
                  class="ant-row ant-row-end"
                >
                  <a
                    href="/login"
                  >
                    resetPassword.redirectToLogin
                  </a>
                </div>
                <div
                  class="ant-row"
                >
                  <button
                    class="ant-btn ant-btn-primary ant-btn-lg sc-dIsUp izwWel"
                    type="submit"
                  >
                    <span
                      aria-label="login"
                      class="anticon anticon-login"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="login"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <defs>
                          <style />
                        </defs>
                        <path
                          d="M521.7 82c-152.5-.4-286.7 78.5-363.4 197.7-3.4 5.3.4 12.3 6.7 12.3h70.3c4.8 0 9.3-2.1 12.3-5.8 7-8.5 14.5-16.7 22.4-24.5 32.6-32.5 70.5-58.1 112.7-75.9 43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 32.6 32.5 58.1 70.4 76 112.5C865.7 417.8 875 464.1 875 512c0 47.9-9.4 94.2-27.8 137.8-17.8 42.1-43.4 80-76 112.5s-70.5 58.1-112.7 75.9A352.8 352.8 0 01520.6 866c-47.9 0-94.3-9.4-137.9-27.8A353.84 353.84 0 01270 762.3c-7.9-7.9-15.3-16.1-22.4-24.5-3-3.7-7.6-5.8-12.3-5.8H165c-6.3 0-10.2 7-6.7 12.3C234.9 863.2 368.5 942 520.6 942c236.2 0 428-190.1 430.4-425.6C953.4 277.1 761.3 82.6 521.7 82zM395.02 624v-76h-314c-4.4 0-8-3.6-8-8v-56c0-4.4 3.6-8 8-8h314v-76c0-6.7 7.8-10.5 13-6.3l141.9 112a8 8 0 010 12.6l-141.9 112c-5.2 4.1-13 .4-13-6.3z"
                        />
                      </svg>
                    </span>
                    <span
                      class="sc-jrsJWt gqswSH"
                    >
                      resetPassword.actionButton
                    </span>
                  </button>
                </div>
                <div
                  class="ant-row ant-row-center ant-row-middle"
                >
                  <a
                    href="/signup"
                  >
                    common.notRegistered
                  </a>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
